storage/src/vespa/storage/distributor/statecheckers.cpp

```diff
@@ -861,11 +862,18 @@ SynchronizeAndMoveStateChecker::check(StateChecker::Context& c)
         IdealStateOperation::UP op(
                 new MergeOperation(BucketAndNodes(c.getBucket(), result.nodes()),
                                    c.distributorConfig.getMaxNodesPerMerge()));
-        op->setPriority(result.priority());
         op->setDetailedReason(result.reason());
-        MaintenancePriority::Priority schedPri(
-                result.needsMoveOnly() ? MaintenancePriority::LOW
-                                       : MaintenancePriority::MEDIUM);
+        MaintenancePriority::Priority schedPri;
+        if (c.getBucketSpace() == document::FixedBucketSpaces::default_space()) {
+            schedPri = (result.needsMoveOnly() ? MaintenancePriority::LOW
+                                               : MaintenancePriority::MEDIUM);
+            op->setPriority(result.priority());
+        } else {
+            // Since the default bucket space has a dependency on the global bucket space,
+            // we prioritize scheduling of merges to global buckets over those for default buckets.
+            schedPri = MaintenancePriority::HIGH;
+            op->setPriority(c.distributorConfig.getMaintenancePriorities().mergeGlobalBuckets);
```

Comment:
This might trigger another kind of starvation if new global documents are fed to a cluster where a node is full.

Reply:
After having looked at the maintenance scheduler, this doesn't seem to be a problem. Buckets are added to a bucket priority database by a maintenance scanner, and the scheduler just picks buckets from that bucket priority database. If a maintenance operation fails then it is not retried until the maintenance scanner checks the bucket again.